controllers/volume/v1alpha2/volume_controller.go

```diff
@@ -95,15 +97,57 @@ func AddToManager(ctx *context.ControllerManagerContext, mgr manager.Manager) er
 		return err
 	}
 
-	// Watch for changes for PersistentVolumeClaim, and enqueue VirtualMachine which is the owner of PersistentVolumeClaim.
-	err = c.Watch(source.Kind(mgr.GetCache(), &corev1.PersistentVolumeClaim{}),
-		handler.EnqueueRequestForOwner(
-			mgr.GetScheme(),
-			mgr.GetRESTMapper(),
-			&vmopv1.VirtualMachine{},
-			handler.OnlyControllerOwner()))
-	if err != nil {
-		return err
+	// Watch for changes for PersistentVolumeClaim, and enqueue VirtualMachine which is the owner
+	// of PersistentVolumeClaim. We only need to watch PVCs because of instance storage.
+	if pkgconfig.FromContext(ctx).Features.InstanceStorage {
+		// Instance storage isn't enabled in all envs and is not that commonly used. Avoid the
+		// memory and CPU cost of watching PVCs until we encounter a VM with instance storage.
+
+		// PVC label we set in createInstanceStoragePVC().
+		isPVCLabels := metav1.LabelSelector{
+			MatchLabels: map[string]string{constants.InstanceStorageLabelKey: "true"},
+		}
+		labelSelector, err := metav1.LabelSelectorAsSelector(&isPVCLabels)
+		if err != nil {
+			return err
+		}
+
+		var pvcWatchStarted bool
+		pvcWatchStartedLock := sync.Mutex{}
```

Comment:
nit: These two lines could be rewritten as:

```golang
		var (
			pvcWatchStarted     bool
			pvcWatchStartedLock sync.Mutex
		)
```

I normally see mutexes declared that way.